### PR TITLE
MDM Enhancements

### LINKED ISSUE #532: Add/document MDM option to enforce passcode
@felix-schwarz Do we already have an option for this? How can I search for such stuff?

Related: https://github.com/owncloud/ios-app/issues/2

### PR DESCRIPTION
**Pull request by mneuwert**
## Description
- Passcode lock enforcement via class setting. User can be forced to set-up a passcode when he first starts the app.
- MDM documentation auto-generated

## Related Issue
#4104

## Motivation and Context
Better customisability using MDM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING**](https://github.com/owncloud/ios-app/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docs/modules/ROOT/pages/ios_mdm.adoc [hunk 1]
```diff
@@ -9,15 +9,610 @@
...
+|`0`
+|Maximum message size in bytes (0 corresponds to 'unlimited').
+
+|log.log-format
+|String
+|`text`
+|Options are `text`, `json` or `json-composed`
+|===
+
+Here is an example of an XML spec-file based on AppConfig standard with minimal logging settings allowing to change a log level and disable / enable private information masking:
```
Comment (michaelstingl) [inline docs/modules/ROOT/pages/ios_mdm.adoc:565]: Add an headline?

```suggestion
== AppConfig XML Example

Here is an example of an XML spec-file based on AppConfig standard with minimal logging settings allowing to change a log level and disable / enable private information masking:
```

File: docs/modules/ROOT/pages/ios_mdm.adoc [hunk 2]
```diff
@@ -9,15 +9,610 @@
...
 The configuration is basically a key-value dictionary provided as a `.plist` file.
 
+=== Configurable Settings
+
+ownCloud App implements a mechanism internally called Class Settings which can be derived from different sources:
+
+- Environment variables which e.g. can be set in Xcode for testing. In this case setting keys have to be prepended with _oc:_ prefix.
+- User preferences accessed by the very same API but stored under _org.owncloud.user-settings_ key.
+- Settings dictionary pushed by an MDM Server and accessible using `NSUserDefaults` API under the key _com.apple.configuration.managed_.
+- Default settings defined directly in the app sourcecode.
```
Comment (michaelstingl) [inline docs/modules/ROOT/pages/ios_mdm.adoc:22]: Also mention, that `Branding.plist` now also can provide values to ClassSettings from a `Configuration` section?

(from https://github.com/owncloud/ios-app/pull/797)

Comment (mneuwert) [reply docs/modules/ROOT/pages/ios_mdm.adoc:22]: @michaelstingl just added this

File: ownCloud/Bookmarks/BookmarkViewController.swift [hunk 3]
```diff
@@ -931,17 +931,31 @@ extension BookmarkViewController : OCClassSettingsSupport {
...
+			.bookmarkDefaultURL : [
+				.type 		: OCClassSettingsMetadataType.string,
+				.description	: "The default URL for the creation of new bookmarks.",
+				.category	: "Bookmarks",
+				.status		: OCClassSettingsKeyStatus.supported
+			],
+
+			.bookmarkURLEditable : [
+				.type 		: OCClassSettingsMetadataType.boolean,
+				.description	: "Controls whetehr the server URL in the text field during the creation of new bookmarks can be changed.",
```
Comment (michaelstingl) [inline ownCloud/Bookmarks/BookmarkViewController.swift:953]:
```suggestion
				.description	: "Controls whether the server URL in the text field during the creation of new bookmarks can be changed.",
```

**Review by hosy — APPROVED** [review]

**Review by michaelstingl — APPROVED** [review]

**Pull request merged**